src/main/java/org/trips/service_framework/controllers/BaseCmsController.java

```diff
@@ -0,0 +1,35 @@
+package org.trips.service_framework.controllers;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.trips.service_framework.aop.Authenticate;
+import org.trips.service_framework.dtos.CmsSkuResponse.Sku;
+import org.trips.service_framework.dtos.SkuAttributes;
+import org.trips.service_framework.helpers.CmsHelper;
+import org.trips.service_framework.models.responses.BaseResponse;
+import org.trips.service_framework.services.CmsService;
+
+import java.util.*;
+
+@RequiredArgsConstructor
+public abstract class BaseCmsController<R extends BaseResponse> {
+    protected final CmsService cmsService;
+    protected final CmsHelper cmsHelper;
+
+    protected abstract R createResponse(List<Sku> skuList);
+
+    @Authenticate
+    @RequestMapping(value = "/search", method = RequestMethod.POST)
+    public R skuSearch(@RequestBody Map<String, String> body) {
+        return createResponse(cmsService.skuSearch(body));
+    }
+
+    @Authenticate
+    @RequestMapping(value = "/lookup-or-create-sku", method = RequestMethod.POST)
```

Comment:
rename to `/find-or-create`

Reply:
Initially, I misunderstood, and thinking of that each service would return different response body.

Now after adding all CMS methods in SF, I find that all response bodies only contain SKU lists. In this PR, I’m ensuring consistency by making sure that every method returns either a single SKU or a list of SKUs.

Also that we’ve added all cms methods in SF, I’ve updated the logic. I’ve removed the generic logic, so we no longer need an additional CMS controller; instead, this SF controller bean will be created.